fix: update docs and add code examples (#151)
LibertyDSNP/graph-sdk: docs/examples/java-examples.md

    List<GraphKeyPair> currentKeyPairs {

    return ImportBundles.newBuilder()
      .addBundles(
        ImportBundles.ImportBundle.newBuilder()
          .setDsnpUserId(msaId)
          .setSchemaId(schemaId)
          .addAllKeypairs(currentKeyPairs)
          .build()
      ).build()
}

fun addNewUserGraphKeypair(long msaId, int schemaId, GraphKeyPair newKeypair)
{
    val bundles = getUserDsnpKeyBundles(msaId, schemaId));
    graph.importUserData(bundles);

    val graphActions = Actions.newBuilder().addActions(
      Actions.Action.newBuilder().setAddKeyAction(
        Actions.Action.AddGraphKey.newBuilder()
          .setOwnerDsnpUserId(msaId)
          .setNewPublicKey(newKeyPair.publicKey)
          .build()
      )
    ).build()

    // add new key to graph
    graph.applyActions(graphActions)

    // export graph updates which in this case is only the added new key
    val updates = graph.exportUpdates()

    // ----- PUBLISH TO BLOCKCHAIN ----- //
    // map exported key into chain request
    val itemizedActions: List<ItemAction> = updates.filter { u -> u.hasAddKey() }.map { update ->
      AddItemAction.from(update.addKey.payload.toByteArray())
    }

>     // finally, add the new exported key to blockchain
>     // here we are using `createApplyItemActions` for simplicity but in reality this should be done using
>     // `createApplyItemActionsWithSignature` since user should sign the payload of adding a new graph key with one of
>     // their msa control keys
>     val retVal =
>       frequencyClient.createApplyItemActions(graphOwnerMsaId, publicKeySchemaId, keyContentHash.toBigInteger(), itemizedActions)
>         .join()

Reviewer: ```suggestion
    // finally, add the new exported key to blockchain
    // the user MUST sign the payload of adding a new graph key with one of their msa control keys
    // The full data to sign is available: https://libertydsnp.github.io/frequency/pallet_stateful_storage/types/struct.ItemizedSignaturePayload.html
    val retVal =
      frequencyClient.createApplyItemActionsWithSignature(graphOwnerSigningPublicKey, signatureProof, publicKeySchemaId, keyContentHash.toBigInteger(), itemizedActions)
        .join()
```

Author: Ah, good catch--I forgot the GraphPublicKey schema was `SignatureRequired` in addition to `AppendOnly`

https://maven.pkg.github.com/LibertyDSNP/graph-sdk

and the package is: `io.amplica.graphsdk`

## Prerequisites

Since GraphSDK does not itself interact with the blockchain, for the following examples, we'll make use of a proposed class, `FrequencyService`, that provides an API to Frequency blockchain RPCs and Extrinsics. The structure/syntax for your own blockchain interface may be somewhat different.

## Basic Social Graph Workflow

The following code examples illustrate how to implement various steps in the workflows represented [here](basic-flow.md)

## Code Example

```kotlin
  fun addNewGraphKeyOrReadPublicKeysWorkflow() {
    // Graph keys are x25519 and different from msa control keys which are sr25519 and should not get mixed
    val newKeyPair = Graph.generateKeyPair(GraphKeyType.X25519)
    val msaId = 1L;
    val publicKey = newKeyPair.publicKey

    // ----- SETUP GRAPH ----- //
    // initialize the Graph in memory
    Configuration configuration = Configuration.getMainnet()
    val graph = Graph(configuration)

    // Get the schema ID for the graph keys
    val schemaId = configuration
    .getEnvironment()

>     .getConfig()

Reviewer: Is this right? Feels off to be calling .getConfig on a variable of type Configuration

    Assertions.assertThat(deserializedKeys).isNotNull
    Assertions.assertThat(deserializedKeys.size).isEqualTo(1)
    Assertions.assertThat(deserializedKeys[0].keyId).isEqualTo(0L)
    Assertions.assertThat(deserializedKeys[0].key).isEqualTo(publicKey)
  }

  fun publicGraphUpdateWorkflow() {
    val msaId = 1L // Dummy ID for example

    // ----- FETCH PUBLIC GRAPH FROM CHAIN ----- //
    val configuration = Configuration(Configuration.getMainnet())
    val publicFollowSchemaId = configuration.getSchemaId(ConnectionType.FollowPublic)
    val pages = frequencyClient.getPaginatedStorage(msaId, publicFollowSchemaId).join()

    val bundlePages = pages.map { page ->
      PageData.newBuilder()
        .setPageId(page.page_id)
        .setContentHash(page.content_hash.toInt())
        // substring use is to skip the `0x` in front of the payload
        .setContent(ByteString.fromHex(page.payload.substring(2)))
        .build()
    }
    val bundles = ImportBundles.newBuilder()
      .addBundles(
        ImportBundles.ImportBundle.newBuilder()
          .setDsnpUserId(msaId.toLong())
          .setSchemaId(publicFollowSchemaId)

>           .setDsnpKeys(DsnpKeys.newBuilder().setDsnpUserId(msaId.toLong()).build())

Reviewer: I don't think we need this line for public graph do we?
```suggestion
```

          .addAllPages(bundlePages)
          .build()
      ).build()

    // ----- SETUP GRAPH ----- //
    // initialize the Graph in memory
    val graph = Graph(configuration)
    graph.importUserData(bundles)

    // create the request to add new connections to graph
    val newConnectionMsaId = 100L;
    val graphActions = Actions.newBuilder().addActions(
      Actions.Action.newBuilder().setConnectAction(
        Actions.Action.ConnectAction.newBuilder()
          .setOwnerDsnpUserId(msaId.toLong())
          .setConnection(
            Connection.newBuilder().setDsnpUserId(newConnectionMsaId).setSchemaId(publicFollowSchemaId).build()
          ).build()
      )
    ).build()

    // add new connections to the graph
    graph.applyActions(graphActions)

    // export graph updates which in this case is graph page update
    val updates = graph.exportUpdates()

    // ----- PUBLISH TO BLOCKCHAIN ----- //
    for (update in updates) {
      if (update.hasPersist()) {
        val persist = update.persist
        // finally, update the graph on chain with new connections
        val retVal =
          frequencyClient.createUpsertPage(

>             msaId,
>             publicFollowSchemaId,

Reviewer: My understanding is that all the values come from persist.

```suggestion
            persist.ownerDsnpUserId,
            persist.schemaId,
```

          .addAllKeyPairs(keyPairs)
          .build()
      ).build()

    // ----- SETUP GRAPH ----- //
    // initializing the Graph in memory
    val graph = Graph(configuration)
    graph.importUserData(bundles)

    // creating the request to add new connections to graph
    val graphActions = Actions.newBuilder().addActions(
      Actions.Action.newBuilder().setConnectAction(
        Actions.Action.ConnectAction.newBuilder()
          .setOwnerDsnpUserId(msaId.toLong())
          .setConnection(
            Connection.newBuilder().setDsnpUserId(newConnectionMsaId).setSchemaId(privateFollowSchemaId).build()
          ).build()
      )
    ).build()

    // adding new connections to the graph
    graph.applyActions(graphActions)

    // exporting graph updates which in this case is graph page update
    val updates = graph.exportUpdates()

    // ----- PUBLISH TO BLOCKCHAIN ----- //
    for (update in updates) {
      if (update.hasPersist()) {
        val persist = update.persist
        // finally, updating the graph on chain with new connections
        val retVal =
          frequencyClient.createUpsertPage(
            msaId,

>             privateFollowSchemaId,

Reviewer: persist.[value]

    val graph = Graph(configuration)
    graph.importUserData(bundles)

    // creating the request to add new connections to graph
    val graphActions = Actions.newBuilder().addActions(
      Actions.Action.newBuilder().setConnectAction(
        Actions.Action.ConnectAction.newBuilder()
          .setOwnerDsnpUserId(msaId.toLong())
          .setConnection(
            Connection.newBuilder().setDsnpUserId(newConnectionMsaId1.value.toLong()).setSchemaId(privateFriendshipSchemaId).build()
          )
          // importing the published keys of the connection so the PRID can get calculated
          .setDsnpKeys(connectionDsnpKeys)
          .build()
      )
    ).build()

    // adding new connections to the graph
    graph.applyActions(graphActions)

    // exporting graph updates which in this case is graph page update
    val updates = graph.exportUpdates()

    // ----- PUBLISH TO BLOCKCHAIN ----- //
    for (update in updates) {
      if (update.hasPersist()) {
        val persist = update.persist
        // finally, updating the graph on chain with new connections
        val retVal =
          frequencyClient.createUpsertPage(
            msaId,

>             privateFriendshipSchemaId,

Reviewer: persist.[value]
